homeassistant/components/tami4/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The Tami4Edge integration."""
+from __future__ import annotations
+
+from Tami4EdgeAPI import Tami4EdgeAPI, exceptions
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import API, CONF_REFRESH_TOKEN, COORDINATOR, DOMAIN
+from .coordinator import Tami4EdgeWaterQualityCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up tami4 from a config entry."""
+    refresh_token = entry.data.get(CONF_REFRESH_TOKEN)
+
+    try:
+        api = await hass.async_add_executor_job(Tami4EdgeAPI, refresh_token)
+    except exceptions.RefreshTokenExpiredException as ex:
+        raise ConfigEntryAuthFailed("API Refresh token expired") from ex
```

Comment:
Raising this error requires implementing the reauthentication flow, which should done in a follow up PR. Please revert the change to use `ConfigEntryError`